ibm/data_source_ibm_atracker_routes.go

```diff
@@ -0,0 +1,185 @@
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of this route.",
+			},
+			"routes": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "A list of route resources.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The uuid of this route resource.",
+						},
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of this route.",
+						},
+						"crn": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The crn of this route type resource.",
+						},
+						"version": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The version of this route.",
+						},
+						"receive_global_events": &schema.Schema{
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Whether or not all global events should be forwarded to this region.",
+						},
+						"rules": &schema.Schema{
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The routing rules that will be evaluated in their order of the array.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"target_ids": &schema.Schema{
+										Type:        schema.TypeList,
+										Computed:    true,
+										Description: "The target ID List. Only one target id is supported.",
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMAtrackerRoutesRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	atrackerClient, err := meta.(ClientSession).AtrackerV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	listRoutesOptions := &atrackerv1.ListRoutesOptions{}
+
+	routeList, response, err := atrackerClient.ListRoutesWithContext(context, listRoutesOptions)
+	if err != nil {
+		log.Printf("[DEBUG] ListRoutesWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+
+	// Use the provided filter argument and construct a new list with only the requested resource(s)
+	var matchRoutes []atrackerv1.Route
+	var name string
+	var suppliedFilter bool
+
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+		suppliedFilter = true
+		for _, data := range routeList.Routes {
+			if *data.Name == name {
+				matchRoutes = append(matchRoutes, data)
+			}
+		}
+	} else {
+		matchRoutes = routeList.Routes
+	}
+	routeList.Routes = matchRoutes
+
+	if len(routeList.Routes) == 0 {
+		return diag.FromErr(fmt.Errorf("no Routes found with name %s\nIf not specified, please specify more filters", name))
```

Comment:
here no Routes found with name %s\nIf not specified, please specify more filters this error message is confusing...

Here we have only filter name  and again asking to specifiy more filters

Reply:
Agreed that this is confusing.  I think we took this message text from an existing data source, perhaps `data_source_ibm_cis.go`, but the situation there is different so it was the wrong pattern to follow.

Further, I don't think we can depend on `name` being set here.

I propose that
1) this check gets moved inside `if suppliedFilter` then block, and
2) the message is changed to simply:
> "no Routes found with name %s", name)

Does this sound good?